app/javascript/angular/code/services/_draw_session.js

```diff
@@ -41,10 +42,7 @@ export const drawSession = (sensors, map, heat, note, empty) => {
         drawnObjects.markers.push(marker);
         points.push(measurement);
       });
-
-      (session.notes || []).forEach(function(noteItem, idx) {
-        drawnObjects.noteDrawings.push(note.drawNote(noteItem, idx));
-      });
+      drawnObjects.notes = drawNotes(session.notes || [], map);
```

Comment:
I tried passing the callback instead of the whole `map`, but map uses this.something in too many places. Maybe in the next refactor :)